src/native/libs/System.Security.Cryptography.Native/pal_evp_kem.c

```diff
@@ -77,6 +78,44 @@ int32_t CryptoNative_EvpKemAvailable(const char* algorithm)
     return 0;
 }
 
+int32_t CryptoNative_EvpKemGetName(const EVP_PKEY* pKey, char* algorithm, int32_t* algorithmLength)
```

Comment:
Is this really "KEM"my, or should it be general EVP_PKEY?

Part of me wants to avoid the UTF-8 strcpy here followed by a UTF-16 transcoding.  But it's probably not going to be anyone's bottleneck, so we can ignore it until proven otherwise.

The only answer I've come up with is to make the function return `void*`, tell the marshaller it returns `void*`, then we do something crazy... and I haven't come up with a good "crazy".  So, I definitely don't have a "good" answer, except that we can have two functions, one that returns an enum (reinventing NID, but only for algorithms we care about), and one that returns the string.  So we get the quick strcmp -> int -> is this what I want, and then on an exception path go ahead and ask for and allocate the UTF-16 string.

Reply:
> The only answer I've come up with is to make the function return void*, tell the marshaller it returns void*, then we do something crazy... and I haven't come up with a good "crazy"

I don't like that because it's a `get0`, and after the p/invoke returns and decrements the handle, there is no guarantee the PKey is alive. I guess we could do DangerousAddRef and the handle will end up getting incremented twice.

But okay. We can make some improvements here.

Reply:
Largely that was just me thinking out loud.  We can stick with the strings for now and noodle on some alternatives... and maybe just stick with the strings forever.

Reply:
It's a valid point. I pushed up an alternative idea. It basically does an enum, and drops the string logic entirely. I am not sure it's worth the trouble for a better exception _message_. Especially since we don't know what the buffer size should be. So maybe we can drop the string stuff.

If you liked the old way better, I can revert 9c06810.

Reply:
I was picturing a more grandiose enum with all of the ML-KEMs, all of the ML-DSAs, and RSA/DSA/ECDSA/ECDH.  But it's a start :smile:.  Having expectations per algorithm family does make the strcmp waterfall faster, so maybe this is better than what I was thinking.